src/anndata/_core/file_backing.py

```diff
@@ -174,7 +175,11 @@ def _(x: AwkArray, *, copy: bool = False):
 
 @to_memory.register(Dataset2D)
 def _(x: Dataset2D, *, copy: bool = False):
-    return x.to_memory(copy=copy)
+    with warnings.catch_warnings():
+        warnings.filterwarnings(  # TODO: somehow only do this in tests but still localized
+            "default", r"Dtype inference.*pandas.*deprecated", FutureWarning
+        )
+        return x.to_memory(copy=copy)
```

Comment:
This is an xarray bug I assume. I changed this so our tests don’t break from xarray’s code encountering this, while still running into it if our code triggers it.

No idea how to do this better, as
- this will override if a user silences this warning.
- ignoring it here means that user’s won’t see this (valid) xarray-triggered warning.

Reply:
I figured out how to fix it instead! I’m setting the `_index` column’s dtype manually before converting it to an index.